src/main/java/com/limechain/cli/Cli.java

```diff
@@ -40,6 +43,15 @@ public Cli() {
         this.options = buildOptions();
     }
 
+    @NotNull
+    private static SyncMode parseSyncMode(CommandLine cmd) throws ParseException {
+        try {
+            return SyncMode.valueOf(cmd.getOptionValue(SYNC_MODE, "warp").toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new ParseException("Invalid sync mode provided, valid values - WARP or FULL");
```

Comment:
I don't like this exception being thrown for 2 reasons:

1. It's not ours. I would expect this to be coming from apache somewhere.
2. It's a checked exception meaning it forces us to handle it with `throws` clauses further coupling us with it.

If you are going to throw something here let it be our own unchecked exception.